### PR TITLE
fix: i18n keys

### PR DESCRIPTION
**Pull request by KaKi87**
### What does it do?

It fixes i18n keys in JS that are outdated compared to the English JSON files.

### Why is it needed?

i18n doesn't work when keys don't match a string from the JSON files.

### How to test it?

These changes were automatically generated by the following Bun script that I wrote :

```js
import { $ } from 'bun';
const
  keys = (await Promise.all((await $`find . -type f -name "en.json"`.text()).trim().split('\n').map(i18nPath => (async () => Object.keys((await import(i18nPath)).default))()))).flat(),
  invalidKeys = [];
let replacedKeyCount = 0;
for(const path of (await $`find ./packages -type f -name "*.ts*" -o -name "*.js*"`.text()).trim().split('\n')){
  const content = await Bun.file(path).text();
  let editedTsContent;
  for(const key of new Set([...content.matchAll(new RegExp(/getTranslation[\n ]*?\([\n ]*?['"`]([a-z0-9.-]+?)['"`][\n ]*?\)/gsi))].map(_ => _[1]))){
    if(keys.includes(key)) continue;
    const replacementKey = keys.find(_key => _key.endsWith(key));
    if(replacementKey){
      editedTsContent = (editedTsContent || content).replace(new RegExp(`(?<=['"\`])${key}(?=['"\`])`, 'g'), replacementKey);
      replacedKeyCount++;
    }
    else
      invalidKeys.push({
        path,
        key
      });
  }
  if(editedTsContent)
    await Bun.write(path, editedTsContent);
}
console.log(`
Automatically replaced ${replacedKeyCount} keys.
Remaining invalid keys :
${invalidKeys.map(({ path, key }) => `- ${path}: ${key}`).join('\n')}
`);
```

Which returns the following output as of d100408 :

```
Automatically replaced 126 keys.
Remaining invalid keys :
- ./packages/plugins/i18n/admin/src/components/CMEditViewCopyLocale.tsx: CMEditViewCopyLocale.ModalConfirm.content
- ./packages/core/admin/ee/admin/src/content-manager/pages/ListView/constants.ts: containers.ListPage.table-headers.reviewWorkflows.stage
- ./packages/core/admin/ee/admin/src/content-manager/pages/ListView/constants.ts: containers.ListPage.table-headers.reviewWorkflows.assignee
- ./packages/core/admin/ee/admin/src/content-manager/pages/ListView/constants.ts: containers.ListPage.table-headers.reviewWorkflows.assignee.label
- ./packages/core/admin/admin/src/content-manager/pages/ListView/components/TableCells/Relations.tsx: DynamicTable.relation-loaded
- ./packages/core/admin/admin/src/content-manager/pages/EditSettingsView/components/ModalForm.jsx: containers.SettingPage.editSettings.size.label
- ./packages/core/admin/admin/src/content-manager/components/DynamicZone/DynamicComponent.tsx: components.DynamicZone.more-actions
- ./packages/core/admin/admin/src/content-manager/components/DynamicZone/DynamicComponent.tsx: components.DynamicZone.add-item-above
- ./packages/core/admin/admin/src/content-manager/components/DynamicZone/DynamicComponent.tsx: components.DynamicZone.add-item-below
```

### Related issue(s)/PR(s)

N/A

---

cc. @derrickmehaffy

Thanks

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx [hunk 1]
```diff
@@ -262,7 +262,7 @@ const CloneDragItem = ({ children, dragHandleTopMargin }: CloneDragItemProps) =>
         forwardedAs="div"
         role="button"
         aria-label={formatMessage({
-          id: getTranslation('components.DragHandle-label'),
+          id: getTranslation('content-manager.content-manager.components.DragHandle-label'),
```
Comment (derrickmehaffy) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:265]: This one might not be correct

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx [hunk 2]
```diff
@@ -228,7 +228,7 @@ const DragAndDropElement = ({
             role="button"
             tabIndex={0}
             aria-label={formatMessage({
-              id: getTranslation('components.DragHandle-label'),
+              id: getTranslation('content-manager.content-manager.components.DragHandle-label'),
```
Comment (derrickmehaffy) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:231]: Same might not be correct

File: packages/core/admin/admin/src/content-manager/components/ContentTypeFormWrapper.tsx [hunk 3]
```diff
@@ -335,7 +335,7 @@ const ContentTypeFormWrapper = ({
         trackUsage('didCreateEntry', trackerProperty);
         toggleNotification({
           type: 'success',
-          message: { id: getTranslation('success.record.save') },
+          message: { id: getTranslation('content-manager.content-manager.success.record.save') },
```
Comment (derrickmehaffy) [inline packages/core/admin/admin/src/content-manager/components/ContentTypeFormWrapper.tsx:338]: Same

**Review by derrickmehaffy — CHANGES_REQUESTED** [review]
Script might have broken some correct values

**Comment by KaKi87** [comment]
Good catch, I'll get right on that.

**Comment by KaKi87** [comment]
The `replaceAll` function was called twice when a key appears twice in the same file. I reverted the initial changes, fixed the script and reran it, which also revealed a few more occurences.

File: packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx [hunk 4]
```diff
@@ -228,7 +228,7 @@ const DragAndDropElement = ({
             role="button"
             tabIndex={0}
             aria-label={formatMessage({
-              id: getTranslation('components.DragHandle-label'),
+              id: getTranslation('content-manager.components.DragHandle-label'),
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:231]: this is incorrect, the `getTranslation` function appends `content-manager` to these strings.

Comment (KaKi87) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:231]: Oh, where can I detect this ?

Comment (derrickmehaffy) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:231]: I'm guessing that prefix append is broken then as I was looking at this with him in Discord and setting the keys to override doesn't do anything, it always takes the static default value.

Comment (KaKi87) [reply packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksContent.tsx:231]: I don't know about `content-manager.components.DragHandle-label` but `content-manager.containers.ListPage.displayedFields` at least was indeed stuck with the default value.

**Comment by derrickmehaffy** [comment]
I tested this again with @KaKi87 @joshuaellis and indeed we found where that prefix [should happen](https://github.com/strapi/strapi/blob/develop/packages/core/admin/admin/src/content-manager/utils/translations.ts) but regardless translation key overrides are broken:

image

So this PR might not be valid but the translation key overrides are certainly broken

**Comment by joshuaellis** [comment]
> So this PR might not be valid but the translation key overrides are certainly broken

that's not very good is it? 😦

**Comment by alexandrebodin** [comment]
Hi, I'm going to close this PR as staled, Let us know if you want to pick up this topic again in the future. THank you for your contribution!

**Comment by KaKi87** [comment]
The topic is still an issue : i18n is currently not working, even though this PR doesn't fix it. Thanks

**Comment by alexandrebodin** [comment]
@KaKi87 can you share the corresponding issue link please ? I'm unclear on what is not working exactly

**Comment by KaKi87** [comment]
I didn't meant *issue* as *GitHub issue*, but as *problem*, sorry :sweat_smile: 

As Derrick summarized it well : *translation key overrides are broken*.

**Comment by alexandrebodin** [comment]
I understood :D I'm just asking if you have a corresponding github issue so we can track and eventually fix this ;)

**Comment by KaKi87** [comment]
Ah, there isn't one, sorry

**Comment by alexandrebodin** [comment]
If you don't mind creating one and add some more info or examples the team will be able to triage it at some point 🙏

**Comment by KaKi87** [comment]
Honestly, since v4 will remain out of Strapi's TODO list until v5 becomes stable, and will then remain low after that, I'm not very motivated to fill the template at this time. 😅

**Comment by alexandrebodin** [comment]
No worries it's fine, we have cleaned things in v5 but it's probably not fixing the root issue so it will be opened later during v5

**Pull request closed without merging**